EIPS/eip-6093.md

```diff
@@ -0,0 +1,376 @@
+---
+eip: 6093
+title: Custom errors for ERC tokens
+description: List of Solidity custom errors covering standard token implementations (EIP-20, EIP-721 and EIP-1155)
+author: Ernesto García (@ernestognw), Francisco Giordano (@frangio), Hadrien Croubois (@Amxx)
+discussions-to: https://ethereum-magicians.org/t/eip-6093-custom-errors-for-erc-tokens/12043
+status: Draft
+type: Standards Track
+category: ERC
+requires: 20, 721, 1155
+created: 2022-12-06
+---
+
+## Abstract
+
+This EIP defines a standard set of Solidity custom errors to be used by [EIP-20](./eip-20.md), [EIP-721](./eip-721.md), and [EIP-1155](./eip-1155.md) tokens.
+
+Ethereum applications (DApps) and wallets have historically relied on revert reason strings to show the cause of transaction errors to users. More recent Solidity versions offer rich revert reasons that require error-specific decoding (sometimes referred to as "custom errors"). This EIP defines a standard set of errors designed to give at least the same relevant information as revert reason strings, but in a structured and expected way that clients can implement decoding for.
+
+## Motivation
+
+Since the introduction of Solidity custom errors in v0.8.4, these have provided a way to show failures in a more expresive way with dynamic arguments, while reducing deployment costs.
```

Comment:
```suggestion
Since the introduction of Solidity custom errors in v0.8.4, these have provided a way to show failures in a more expresive and gas efficient manner with dynamic arguments, while reducing deployment costs.
```